lib/ansible/plugins/doc_fragments/action_common_attributes.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2015, Ansible, Inc
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+
+class ModuleDocFragment(object):
+
+    # Standard documentation fragment
+    DOCUMENTATION = r'''
+attributes:
+    action:
+      description: Indicates this has a corresponding action plugin so some parts of the options can be executed on the controller
+      support: no
+    async:
+      description: Supports being used with the ``async`` keyword
+      support: full
+    become:
+      description: Is usable alongside become keywords
+      support: full
+    bypass_host_loop:
+      description: Forces a 'global' task that does not execute per host, cannot be used in non lockstep strategies
+      support: no
+    check_mode:
+      description: Can run in check_mode and return changed status prediction
+      support: no
+    connection:
+      description: Uses the target's configured connection information to execute code on it
+      support: full
+    conditional:
+      description: Will respect the `when` keyword  per item loop or task (when no loop is present)
+      support: full
+    delegation:
+      description: Can be used in conjunction with delegate_to and related keywords
+      support: full
+    diff:
+      description: Will return details on what has changed when in diff is enabled
+      support: no
+    facts:
+      description: Action returns an ``ansible_facts`` dictionary that will update existing host facts
+      support: no
+    loops:
+      description: both ``loop`` and ``with_`` looping keywords will be honored.
+      support: yes
```

Comment:
Allowing people to use either `yes` or `full` seems confusing - is this a boolean field or isn't it?

How about allowing three values: `none`, `partial`, and `full`? That seems clear and it covers all the possibilities I know of.

Reply:
works4me